Source/MediaInfo/Tag/File_Exif.cpp

```diff
@@ -1374,6 +1745,48 @@ void File_Exif::FileHeader_Parse()
         }
         OffsetFromContainer = 10;
     }
+    if (IsMakernote) {
+        if (Element_Size >= 10 && !strncmp((const char*)Buffer + Buffer_Offset, "Apple iOS", 10)) { // the char* contains a terminating \0
+            int16u alignment;
+            Skip_String(10,                                     "Identifier");
+            Skip_XX(2,                                          "(unknown)");
+            Get_C2(alignment,                                   "Alignment");
+            if (alignment == 0x4D4D)
+                LittleEndian = false;
+            else {
+                Reject();
+                return;
+            }
+            currentIFD = Kind_MakernoteApple;
+            Accept();
+            Stream_Prepare(Stream_Image);
+            return;
```

Comment:
I don't like the idea of returning there, is it possible to avoid the "return" here?